### DISCUSSION TITLE
no space before @endif gives error in html element

### DISCUSSION THREAD

**Discussion by MahmudSuleman** [laravel/framework#43719] [Q&A]
Hi, I noticed that when I don't put a space before an @endif I get an error.
code below gives an error.

![image](https://user-images.githubusercontent.com/38923023/184870197-eb2d9a5f-3ac7-4587-8caf-7fa073063c70.png)
The code below works fine
![image](https://user-images.githubusercontent.com/38923023/184870412-71cfd49e-37f3-46fa-a233-de5a16504b68.png)

I will be grateful if someone could explain it to me.

**Comment by aguingand** [comment]
Blade tags requires a space before. For classes you can use the `@class()` directive, see the [documentation](https://laravel.com/docs/9.x/blade#conditional-classes).

**Comment by kaozaza2** [comment]
try `@class(['your-base-class', 'bg-red-300' => $loop->even])`
